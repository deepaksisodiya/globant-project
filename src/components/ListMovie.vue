<template>
  <div>
    <ul>
      <li v-for="(movie, index) in movies" :key="index">
        <p>{{ movie.movieName }}</p>
        <p>{{ movie.genre }}</p>
        <p>{{ movie.rating }}/5</p>
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  name: "ListMovie",
  props: {
    movies: {
      type: Array,
      required: true
    }
  }
};
</script>

<style>
ul {
  background: #cce5ff;
  padding: 20px;
}

ul li {
  background: #3399ff;
  margin: 5px;
}
</style>
