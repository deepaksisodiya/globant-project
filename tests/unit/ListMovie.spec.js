import { shallowMount } from "@vue/test-utils";
import ListMovie from "@/components/ListMovie.vue";

describe("ListMovie.vue", () => {
  it("should be an vue instace", () => {
    const movies = [];
    const wrapper = shallowMount(ListMovie, {
      propsData: { movies }
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("should match snapshot with empty movies list", () => {
    const movies = [];
    const wrapper = shallowMount(ListMovie, {
      propsData: { movies }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should match snapshot with movies list", () => {
    const movies = [
      {
        movieName: "movieName",
        genre: "genre",
        rating: "rating",
        website: "website"
      }
    ];
    const wrapper = shallowMount(ListMovie, {
      propsData: { movies }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
