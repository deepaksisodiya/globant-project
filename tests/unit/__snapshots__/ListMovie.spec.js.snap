// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMovie.vue should match snapshot with empty movies list 1`] = `
<div>
  <ul />
</div>
`;

exports[`ListMovie.vue should match snapshot with movies list 1`] = `
<div>
  <ul>
    <li>
      <p>
        movieName
      </p>
       
      <p>
        genre
      </p>
       
      <p>
        rating/5
      </p>
    </li>
  </ul>
</div>
`;
